### PR TITLE
Prefab Component

### LINKED ISSUE #7240: : Refactor AssetComponent to AssemblyComponent
- Rename all relevant types, functions, systems to reflect refactor.
- Clean up node editor, create "AssemblyInput" react component to drag and drop 'xre.gltf' files onto

### PR DESCRIPTION
**Pull request by dinomut1**
Refactor AssetComponent into PrefabComponent, improve ECS conversion to/from .xre.gltf
closes #7240

### CODE REVIEW

File: packages/engine/src/scene/systems/SceneObjectSystem.tsx [hunk 1]
```diff
@@ -57,16 +58,17 @@ export function setupObject(obj: Object3DWithEntity) {
...
         const onlyEmmisive = prevMaterial.emissiveMap && !prevMaterial.map
+        const prevMatEntry = unregisterMaterial(prevMaterial)
         prevMaterial.dispose()
         child.material = new MeshBasicMaterial().copy(prevMaterial)
         child.material.color = onlyEmmisive ? new Color('white') : prevMaterial.color
         child.material.map = prevMaterial.map ?? prevMaterial.emissiveMap
-
         // todo: find out why leaving the envMap makes basic & lambert materials transparent here
         child.material.envMap = null
+        registerMaterial(child.material, prevMatEntry.src)
```
Comment (HexaField) [inline packages/engine/src/scene/systems/SceneObjectSystem.tsx:71]: This is causing MaterialNotFound errors in locations

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/scene/systems/SceneObjectSystem.tsx [hunk 2]
```diff
@@ -57,16 +58,17 @@ export function setupObject(obj: Object3DWithEntity) {
   const mesh = obj as any as Mesh<any, any>
   mesh.traverse((child: Mesh<any, any>) => {
     if (child.material) {
-      if (isHMD && ExpensiveMaterials.has(child.material.constructor)) {
+      if (!Engine.instance.isEditor && !isHMD && ExpensiveMaterials.has(child.material.constructor)) {
```
Comment (HexaField) [inline packages/engine/src/scene/systems/SceneObjectSystem.tsx:61]: incorrect inversion of isHMD (should be `isHMD` not `!isHMD`)

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/scene/systems/SceneObjectSystem.tsx (hunk 2 above)
Comment (dinomut1) [reply packages/engine/src/scene/systems/SceneObjectSystem.tsx:61]: good catch

File: packages/common/src/utils/CommonKnownContentTypes.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@
  * @type {Object}
  */
 export const CommonKnownContentTypes = {
-  xre: 'asset/xre',
+  xre: 'prefab/xre',
   gltf: 'model/gltf',
```
Comment (speigg) [inline packages/common/src/utils/CommonKnownContentTypes.ts:8]: So, are prefab files a custom file type we are introducing? I thought they were just gltf files?

Comment (dinomut1) [reply packages/common/src/utils/CommonKnownContentTypes.ts:8]: they are the ecs represented in a gltf file

**Pull request closed without merging**